apps/prairielearn/src/middlewares/authzCourseOrInstance.ts

```diff
@@ -418,18 +346,14 @@ export async function authzCourseOrInstance(req: Request, res: Response) {
     course_instance_id: req.params.course_instance_id || null,
     is_administrator,
     allow_example_course_override: false,
-    ip: req.ip,
+    ip: req.ip || null,
     req_date,
     req_mode: res.locals.authz_data.mode,
     req_course_role: req.cookies.pl2_requested_course_role || null,
     req_course_instance_role: req.cookies.pl2_requested_course_instance_role || null,
   };
 
-  const effectiveAuthzData = await sqldb.queryOptionalRow(
-    sql.select_authz_data,
-    effectiveParams,
-    SelectAuthzDataSchema,
-  );
+  const effectiveAuthzData = await selectAuthzData(effectiveParams);
```

Comment:
Nit: the separate "params" object is kind of an old pattern - can we just pass an object directly here?